web_advanced_search/static/src/js/human_domain.js

```diff
@@ -1,69 +1,58 @@
 /* Copyright 2018 Tecnativa - Jairo Llopis
+ * Copyright 2020 Tecnativa - Alexandre Díaz
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-odoo.define("web_advanced_search.human_domain", function(require) {
+odoo.define("web_advanced_search.human_domain", function() {
     "use strict";
 
-    var DomainSelector = require("web.DomainSelector");
-
-    var join_mapping = {
+    const join_mapping = {
         "&": _(" and "),
         "|": _(" or "),
         "!": _(" is not "),
     };
 
-    // HACK I should extend classes, but they are not exposed
-    // TODO Remove file when merged https://github.com/odoo/odoo/pull/25922
-    var human_domain_methods = {
+    const human_domain_methods = {
         DomainTree: function() {
-            var human_domains = [];
-            _.each(this.children, function(child) {
+            const human_domains = [];
+            _.each(this.children, child => {
                 human_domains.push(human_domain_methods[child.template].apply(child));
             });
-            return _.str.sprintf(
-                "(%s)",
-                human_domains.join(join_mapping[this.operator])
-            );
+            return `(${human_domains.join(join_mapping[this.operator])})`;
         },
 
         DomainSelector: function() {
-            var result = human_domain_methods.DomainTree.apply(this, arguments);
+            const result = human_domain_methods.DomainTree.apply(this, arguments);
             // Remove surrounding parenthesis
             return result.slice(1, -1);
         },
 
         DomainLeaf: function() {
-            var chain = [],
-                operator = this.operator_mapping[this.operator],
-                value = _.str.sprintf('"%s"', this.value);
+            const chain = [];
+            let operator = this.operator_mapping[this.operator],
+                value = `"${this.value}"`;
             // Humanize chain
-            this.chain.split(".").forEach(function(element, index) {
+            const chain_splitted = this.chain.split(".");
+            const len = chain_splitted.length;
+            for (let x = 0; x < len; ++x) {
+                const element = chain_splitted[x];
```

Comment:
Is this worth it? Why not just:

```js
             this.chain.split(".").forEach((element, index) => {
```

Almost like before, but you avoid having to manually bind the function.